### PR TITLE
Modernize all the test cases

### PR DESCRIPTION
**Pull request by felixdivo**
This modernizes all python files in `test/`. I basically ran `2to3` on it and the went over them by hand to correct or add some changes manually.

### CODE REVIEW

**Comment by felixdivo** [comment]
This requires #594 to be merged first.

File: test/back2back_test.py [hunk 1]
```diff
@@ -5,8 +5,6 @@
 This module tests two virtual buses attached to each other.
 """
 
-from __future__ import absolute_import, print_function
```
Comment (hardbyte) [inline test/back2back_test.py]: :rocket: we are in the future.

File: test/test_kvaser.py [hunk 2]
```diff
@@ -8,10 +8,7 @@
 import time
 import logging
 import unittest
-try:
-    from unittest.mock import Mock, patch
-except ImportError:
-    from mock import patch, Mock
```
Comment (hardbyte) [inline test/test_kvaser.py]: We can remove the `mock` dependency if not already done.

Comment (felixdivo) [reply test/test_kvaser.py]: Done. As well as the future dependency. Six will be removed separately, I opened an issue for it.

**Pull request merged**